tools.go

```diff
@@ -0,0 +1,5 @@
+//go:build tools
+
+package winio
+
+import _ "golang.org/x/tools/cmd/stringer"
```

Comment:
Is the goal here to control what version of `golang.org/x/tools/cmd/stringer` is used in `go generate` via the `go.mod` file?

Reply:
yup, I have a PR somewhere to add something similar to hcsshim for `github.com/josephspurrier/goversioninfo/cmd/goversioninfo` and `github.com/Microsoft/go-winio/tools/mkwinsyscall`.
But also, I dont know the full details, but if its not in go.mod, then `go run golang.org/x/tools/cmd/stringer` (and therefore `go generate`) sometimes fails for me

Reply:
What failure do you get? I haven't heard of that before.

Reply:
```pwsh
> go generate .\pkg\guid\
no required module provides package golang.org/x/tools/cmd/stringer; to add it:
        go get golang.org/x/tools/cmd/stringer
pkg\guid\guid.go:17: running "go": exit status 1
```

Though running `go get` adds it to `go.mod`, but with an indirect flag:
```
golang.org/x/tools v0.1.12 // indirect
```

So I guess we can manually track tools via `go get`, but I went the go tools route following the [Go recommendation](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module).

Reply:
Ah cool, I wasn't aware of this recommendation. That link would be a good thing to include in your commit description.